MyStore/Settings.cpp

```diff
@@ -0,0 +1,38 @@
+#include "Settings.h"
+
+Settings::~Settings()
+{
+	if (m_instance != nullptr)
+		delete m_instance;
+}

```

Comment:
Will this ever be called? You don't have any Settings objects on the stack, so it seems like this destructor will never be called and as a result m_instance won't be freed.